app/models/course/discussion/post.rb

```diff
@@ -1,5 +1,16 @@
 class Course::Discussion::Post < ActiveRecord::Base
   acts_as_forest order: :created_at
 
+  after_initialize :set_topic, if: :new_record?
+  validate :validate, if: 'parent.present?'
+
   belongs_to :topic, inverse_of: :posts
+
+  def set_topic
+    self.topic = parent.topic if parent
+  end
+
+  def validate
```

Comment:
check the naming convention and give this a better name ? 

Also need specs for this.